simpledotorg/simple-server
# Add Hypertension report to Progress tab

**Story card:** [sc-7581](https://app.shortcut.com/simpledotorg/story/7581/add-new-reports-card-and-create-hypertension-report)

## Because

We will show a full hypertension report in the progress tab that includes the following data:
* Total assigned patients (current count)
* Total registered patients trend (last 6 months)
* Monthly follow-up patients trend (last 6 months)
* BP controlled trend (last 6 months)
* BP not controlled trend (last 6 months)
* Missed visits trend (last 6 months)
* Quarterly cohort reports (last 4 quarters)

## This addresses
**Original progress report home page**
![image](https://user-images.githubusercontent.com/16785131/159541956-0f205138-f3bf-4c4f-b059-0cfd52af7303.png)

**New progress report home page**
![image](https://user-images.githubusercontent.com/16785131/159542035-35f5a3d9-4a72-4a38-afff-f3dea4c52cb7.png)

**New progress hypertension report**
![image](https://user-images.githubusercontent.com/16785131/159542073-6be0403c-bab5-4cb5-b467-569e3fb9fb0f.png)

## Test instructions
**1. Make sure the original home page works IF the Flipper flag is not enabled:**
* Go to a facility progress report page (Reports > Select a facility report from the search bar > Progress report).
* Change your browser to a mobile view.
* Make sure the "Today" card shows data for past days.
* Make sure the "Registered patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Follow-up patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Hypertension controlled" card works. Tap the chart area and make sure the month labels show. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Click the "Cohort reports" button and make sure the cohort report page shows up. When you click out of the page, make sure the home page appears.
* Make sure the "Achievements" card shows total follow-up and total registrations values that match the totals shown in the cards above. Also make sure the badges match the values. For example, if there are 27 follow-ups, there should be 1 completed badge and 4 empty badges.

**2. Make sure the new home page and hypertension report show IF the Flipper flag, `new_progress_tab` is enabled:**
* Go to Settings > Flipper and add/enable the `new_progress_tab` flag.
* Go to a facility progress report page (Reports > Select a facility report from the search bar > Progress report).
* Change your browser to a mobile view.
* Make sure the "Today" card shows data for the past days.
* Make sure the "Registered patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Follow-up patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the old "Hypertension controlled" card and "Cohort reports" button aren't showing.
* Make sure the new "Reports" card is showing and that the "Hypertension" button is active and the "Diabetes report" button is gray and says "Coming soon"
* Click the "Hypertension report" button, and scroll down the page. Make sure the charts look good. Make sure the "Patient treatment outcomes" section help icon and chart data point tooltips work and show the right information/data. Make sure the "Cohort report" bar charts work and show the tooltips correctly above each bar.
